### DISCUSSION TITLE
rest-api enable fail

### DISCUSSION THREAD

**Discussion by steven-ld** [Keyfactor/ejbca-ce#754] [Q&A] [answered]
I turned on the interface of the v1 version, but it doesn't seem to have any effect.

![image](https://github.com/user-attachments/assets/d44cc197-d493-49a2-b00f-6cf7c4cb3328)
![image](https://github.com/user-attachments/assets/c787b0ee-43bd-436a-be6a-badc6342ca75)
![image](https://github.com/user-attachments/assets/872a039e-f8e6-4afd-ac14-843f56db05b2)
![image](https://github.com/user-attachments/assets/42e21cbb-afed-4518-80f3-1082bdc351d9)

**Comment by primetomas** [comment]
What version of EJBCA?

**Reply by steven-ld** [reply]
I'm using docker image  “keyfactor/ejbca-ce:8.3.1”
![image](https://github.com/user-attachments/assets/a855cff5-3b98-411b-abe7-6459924a870d)
This is my startup command.
```
docker run -it -d --name=ejbca-8.3.1  -p 80:8080 -p 443:8443 -p 8080:8080 -p 8443:8443  -h localhost -e TLS_SETUP_ENABLED="true" keyfactor/ejbca-ce:8.3.1
```

**Comment by primetomas** [comment]
EJBCA 9.0 is out. You can upgrade to this.

Try to check, save, uncheck, save the custom header checkbox. That did it for me. If that works for you it may be a display issue in the UI.

**Reply by steven-ld** [reply]
I upgraded to version 9.0.0 and tried the operation you said. It does take effect in the browser, but it still cannot be called in ejbca-sdk-java. It prompts the use of the acquisition request. I printed the method used by the client as POST, but rest-api said it was get and rejected the request.
![image](https://github.com/user-attachments/assets/6b29ec43-4c78-4af9-9eb2-717f044cef9e)
![image](https://github.com/user-attachments/assets/ea2c20e8-284a-4972-870d-35f1d4d8c9da)

**Reply by steven-ld** [reply]
I think I know what the problem is. I will return the same error when I use postman without adding any certificate. I think it's because of the certificate. After ejbca started, I got a SuperAdmin.p12 file and a password. At present, I have used two pem files in the SDK, which I extracted from the SuperAdmin.p12 file through the following command.
```
## Step1:Extract certificate
openssl pkcs12 -in Superadmin.p12 -nokeys -out certificate.pem

#Step2:Extract the key, but set a password for this key at the same time.
openssl pkcs12 -in SuperAdmin.p12  -nocerts -out key.pem

#Step3: Remove the password of the previous step
openssl rsa -in key.pem -out key_nopass.pem

```
I don't know how to get the certificates and keys mentioned in sdk now.

**Reply by primetomas** [reply]
Yes you need client certificate authentication.
Here is a couple of good resources for remotely calling the REST API. 
https://github.com/Keyfactor/ejbca-easy-rest-client
https://github.com/Keyfactor/ejbca-go-client-sdk

Or just use cURL:
https://docs.keyfactor.com/ejbca/latest/ejbca-rest-interface#id-(9.1.1)EJBCARESTInterface-IntegratingwiththeRESTAPI

**Reply by steven-ld** [reply]
You are right. i can check how it is implemented in two other ways, but the fundamental reason is how to obtain the two certificates java-sdk. If it is based on easy-rest-client, there will be a lot of development tasks. Does it feel that java-sdk has stopped updating?

**Reply by primetomas** [reply]
I just haven't used the java client myself. I guess you mean this one?
https://github.com/Keyfactor/ejbca-java-client-sdk

You have to see what error you get there and configure accordingly. getting the client key and cert from superadmin.p12 should work fine.
I notice that the example doesn't set setSslCaCert which may be needed for the TLS connection, unless you set setVerifyingSsl(false). 

I just haven't tested that myself.

**Reply by steven-ld** [reply, marked as answer]
Well, I can only explore slowly. setVerifyingSsl(false) has been set in the demo. Except for the interface return exceptions, there is no abnormality in sdk work，thank you very much

**Comment by Nika250** [comment]
You are right. I can check, thanks.
